### DISCUSSION TITLE
What should I expect the output for ESM format to look like? It doesn't look right to me

### DISCUSSION THREAD

**Discussion by ShaneYu** [web-infra-dev/rslib#161] [Q&A] [answered]
Hello,

I have been playing around using RsLib to build a React components library, where I am outputting both CJS and ESM artifacts. However I've noticed that the ESM output only contains `*.d.mts` files and not any `*.mjs` files and so wanted to check if the output is actually correct?

![image](https://github.com/user-attachments/assets/b50dbfa1-c8d1-4e20-98c4-0a13ec2a425a)

If you take a look at the attached screenshot above, you can see that the CJS output has included index.js and index.d.ts as expected, however ESM has only put out `index.d.mts` which I don't feel is correct.

Please can you confirm if this is correct? My gut feeling was that I would see a `index.mjs` and a `index.d.ts` or `index.d.mts` file. Browsing each of the ESM files I can only see type definitions like declares and types with import/export statements, I am not seeing any actual javascript code to represent the button and alert components like you see within the CJS output.

FYI, the code I am using for the rslib.config.json file is based on the examples in this repository, however I had to set the source.entry config options for it to work, otherwise rslib fails saying it cannot find the entry file; I wanted to have multiple entry points anyway, so this isn't an issue; but is there it differs from the example in this repo.

**Comment by fi3ework** [comment, marked as answer]
Hi, thanks for try Rslib before it reaching a stable version. I think this might be a bug in Rslib, but we need more information to help debug, and tracking this issue through an issue rather than a discussion would be better, could you go ahead and open an issue through https://github.com/web-infra-dev/rslib/issues/new/choose, we'll follow up. Thanks.

**Reply by ShaneYu** [reply]
No worries, I will get that raised for you momentarily. 😁

**Comment by ShaneYu** [comment]
I have raised the bug here: https://github.com/web-infra-dev/rslib/issues/162

I've pushed up a temp GitHub repository to help with the replication of the issue(s).
